compose/config/config.py

```diff
@@ -188,10 +217,46 @@ def load(config_details):
 
     Return a fully interpolated, extended and validated configuration.
     """
+    version = get_config_version(config_details)
+    processed_files = []
+    for config_file in config_details.config_files:
+        processed_files.append(
+            process_config_file(config_file, version=version)
+        )
+    config_details = config_details._replace(config_files=processed_files)
+
+    if not version or isinstance(version, dict):
+        service_dicts = load_services(
+            config_details.working_dir, config_details.config_files
+        )
```

Comment:
Since this function takes both parts of a `config_details`, I would just have it accept a single arg of `config_details`  instead of splitting it here. What do you think?

```python
service_dicts = load_services(config_details)
```